packages/core/useFetch/index.ts

```diff
@@ -481,7 +481,13 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
         if (!payloadType && unref(payload) && Object.getPrototypeOf(unref(payload)) === Object.prototype)
           config.payloadType = 'json'
 
-        return shell as any
+        return {
+          ...shell,
+          then(onFulfilled: any, onRejected: any) {
+            return waitUntilFinished()
+              .then(onFulfilled, onRejected)
+          },
+        } as any
```

Comment:
maybe like this
```suggestion
        } as UseFetchReturn<T> & PromiseLike<UseFetchReturn<T>>
```

Reply:
Yeah fair suggestion, I just copied the implementation of `setType` below :) https://github.com/vueuse/vueuse/pull/1571/files#diff-0db62ed388d474bc80cd69e2ff1593cd6c9719b8794d8f88459a1704e8da25b5L508